### PR TITLE
Add subtype field for state blocks in RPC block_info, blocks_info

### PR DESCRIPTION
**Pull request by SergiySW**
https://github.com/nanocurrency/nano-node/issues/809

### CODE REVIEW

File: nano/node/rpc.cpp [hunk 1]
```diff
@@ -834,6 +834,31 @@ void nano::rpc_handler::block_info ()
...
+				{
+					response_l.put ("subtype", "send");
+				}
+				else
+				{
+					if (block->link ().is_zero ())
+					{
+						response_l.put ("subtype", "change");
+					}
+					else if (balance == previous_balance && !node.ledger.epoch_link.is_zero () && node.ledger.is_epoch_link (block->link ()))
```
Comment (cryptocode) [inline nano/node/rpc.cpp]: We don't seem to be checking if epoch_link is zero anywhere else but subtype checking - is that necessary?

Comment (SergiySW) [reply nano/node/rpc.cpp]: We check it in ledger.cpp processing as well

File: nano/node/rpc.cpp [hunk 2]
```diff
@@ -941,7 +966,30 @@ void nano::rpc_handler::blocks_info ()
 						block->serialize_json (contents);
 						entry.put ("contents", contents);
 					}
-
+					// Subtype check
+					if (block->type () == nano::block_type::state)
+					{
```
Comment (wezrule) [inline nano/node/rpc.cpp]: This block (20+ lines) is identical to that in `block_info ()` can it be refactored into a separate function?

**Comment by severalif** [comment]
I think it would be useful to include this information in the RPC callback as well.

**Comment by SergiySW** [comment]
@severalif callback is updated as well

**Comment by renesq** [comment]
Callback didn't have a subtype field, because a state block can have multiple purposes at the same time, so the "is_send" flag was introduced instead, which is less misleading. I don't think the "subtype" makes any sense here at all, because it will give you no further info that you could get by having the "is_send" flag and looking at the callback content ("amount" and "is_send"). It will just add more traffic.

**Comment by SergiySW** [comment]
@renesq unfortunantely some services not willing to understand many things in Nano. The simpler, the better

**Comment by renesq** [comment]
In that case, revise the current block concept as a whole. Details in #1832

**Comment by SergiySW** [comment]
Yes, several heavy issues exists with state blocks design, but making new types again with a lot of legacy types support

**Pull request merged**